### PR TITLE
TDL-7361: Make Chunk Size Configurable

### PR DESCRIPTION
**Pull request by hpatel41**
# Description of change
TDL-7361: Make Chunk Size Configurable
- Updated the code to make `chunk_size` configurable

# Manual QA steps
 - 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tests/unittests/test_chunk_size.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        # Verify we raise ValueError
+        with self.assertRaises(ValueError) as e:
+            ReportEmailActivity({}, client, {**self.config, **test_data}, Catalog(), [], [])
+
+        # Verify the error message
+        self.assertEqual(str(e.exception), 'Please provide a valid integer value for the chunk_size parameter.')
+
+        # Verify we get the logger when Negative and Zero(0) chunk_size is passed in the config
+        if expected_data[0]:
+            mocked_logger_info.assert_called_with(expected_data[1])
```
Comment (savan-chovatiya) [inline tests/unittests/test_chunk_size.py:96]: Keep scenario-specific test suites in two different test cases rather than using the if condition.

**Review by umangagrawal-crest — COMMENTED** [review]
Update the README file

**Comment by harshpatel4crest** [comment]
> Update the README file

Updated the README file with the `chunk_size` config param.

File: tests/unittests/test_chunk_size.py (hunk 1 above)
Comment (harshpatel4crest) [reply tests/unittests/test_chunk_size.py]: Updated the test case to separate tests.

**Review by dbshah1212 — APPROVED** [review]

**Review by savan-chovatiya — APPROVED** [review]

**Review by Vi6hal — COMMENTED** [review]
approved

**Pull request merged**